### DISCUSSION TITLE
Using isopod alongside another app (outside of MOOSE)

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#22168] [Q&A General] [answered]
Hi,

I'd like to call some of Isopod's objects in an experimental Ferret branch. If I clone it along side moose in the same folder (/projects/moose and /projects/ferret and projects/isopod), how do I modify Ferret's makefile for this?

Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

Easy solution is to copy all of Ferret into isopod. Just copy all the src and headers, and re-register all objects. 

Other solutions:
- try to get dynamic loading to work. This is the (limited) docs for it
https://mooseframework.inl.gov/moose/source/actions/DynamicObjectRegistrationAction.html

- build a combined other application (with both other apps and moose as submodules).
There s advice on this here:
https://github.com/idaholab/moose/discussions/21980

Guillaume

**Comment by permcody** [comment]
John, do you have access to one of the other applications that combines more than one sub-application? Generally you can add about 3-4 lines to your Makefile to pull this off. Something like the following inserted before the final executable block:

```
APPLICATION_DIR    := <directory>
APPLICATION_NAME   := <application name>
include            $(FRAMEWORK_DIR)/app.mk
```

**Reply by GiudGiud** [reply]
You also need to modify the base/XXXApp.C to register the objects

**Comment by mangerij** [comment]
Thanks guys!

**Reply by GiudGiud** [reply]
also in case no one mentioned this before, the optimization module in a PR in MOOSE right is isopod

**Reply by GiudGiud** [reply]
https://github.com/idaholab/moose/pull/21887
